### ISSUE TITLE
Clang's static analyzer should complain about reflexive assignment

### ISSUE THREAD

**Issue by ryao** [llvm/llvm-project#61033]
In #c on the libera IRC network, someone said:

> `int fd = fd;` this cost me about half an hour today

I wrote a small test program:

```
int main() {
int a = 0;
a = a;
return 0;
}
```

Both `clang --analyze /tmp/same.c` and `clang-tidy  /tmp/same.c --` do not report any issues. However, copy and pasting that into PVS Studio via godbolt (an extra new line at the end might be needed) generates a warning:

https://pvs-studio.com/en/pvs-studio/godbolt/

https://pvs-studio.com/en/docs/warnings/v570/

It would be nice if Clang's static analyzer would complain about this as well since anything doing this is probably doing something wrong.

**Comment by firewave** [comment]
It is a compiler and not an analyzer warning:
```
<source>:3:7: warning: explicitly assigning value of variable of type 'int' to itself [-Wself-assign]
    a = a;
    ~ ^ ~
```

https://godbolt.org/z/GqqKejdMT

**Comment by ryao** [comment]
I did not think to check for that. My apologies for the noise.
